### PR TITLE
RD-686 Make BP validator use new workflow

### PR DESCRIPTION
**Pull request by glukhman**

### CODE REVIEW

File: tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                      '.com/blueprints/the-not-blueprint-master.zip'
+        self.client.blueprints.validate(
+            archive_url,
+            blueprint_id)
+        temp_blueprint_id = self._assert_blueprint_validation_message(
+            blueprint_id, 'Blueprint validated.')
+        self._assert_cleanup(temp_blueprint_id)
+
+    def test_blueprint_validate_invalid_blueprint(self):
+        blueprint_id = 'bp-bad-schema'
```
Comment (YoniItzhak) [inline tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py:49]: This is the same flow as here https://github.com/cloudify-cosmo/cloudify-manager/pull/2533/files#diff-6a28a93db36b5a4785a60c49c9b49d01a993f8f666d75a5f82c2e7ec015a4afdR27

You can take it out to a function.

File: tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        self.client.blueprints.validate(
+            resource('dsl/{}'.format(blueprint_filename)),
+            entity_id=blueprint_id
+        )
+        temp_blueprint_id = self._assert_blueprint_validation_message(
+            blueprint_id, "Failed parsing blueprint - 'foo' is not in schema.")
+        self._assert_cleanup(temp_blueprint_id)
+
+    def _assert_blueprint_validation_message(self, blueprint_id, message):
+        execution = [
```
Comment (YoniItzhak) [inline tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py:60]: Can `client.blueprints.validate()` return the execution ID?

Comment (glukhman) [reply tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py:60]: See discussion here: https://github.com/cloudify-cosmo/cloudify-cli/pull/1175#discussion_r554544220

File: tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py (hunk 1 above)
Comment (glukhman) [reply tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py:44]: done

File: rest-service/manager_rest/upload_manager.py [hunk 3]
```diff
@@ -679,137 +670,62 @@ def _process_plugins(cls, file_server_root, blueprint_id):
...
+            visibility=None,
+            state=BlueprintUploadState.VALIDATING
+        ))
 
-        app_file_name = cls._extract_application_file(
-            file_server_root, app_dir, args.application_file_name)
+        if not blueprint_url:
+            try:
+                self.upload_archive_to_file_server(blueprint_id)
+            except Exception as e:
```
[the remark this answers is not in the document]
Comment (glukhman) [reply rest-service/manager_rest/upload_manager.py]: done

File: rest-service/manager_rest/rest/resources_v3_1/blueprints.py [hunk 4]
```diff
@@ -167,6 +169,18 @@ def patch(self, blueprint_id, **kwargs):
         sm = get_storage_manager()
         blueprint = sm.get(models.Blueprint, blueprint_id)
 
+        # if finished blueprint validation - cleanup DB entry
+        # and uploaded blueprints folder
+        if blueprint.state == BlueprintUploadState.VALIDATING:
+            uploaded_blueprint_path = '{0}/{1}/{2}/{3}'.format(
```
[the remark this answers is not in the document]
Comment (glukhman) [reply rest-service/manager_rest/rest/resources_v3_1/blueprints.py]: done

File: rest-service/manager_rest/rest/resources_v3_1/blueprints.py [hunk 5]
```diff
@@ -166,6 +168,18 @@ def patch(self, blueprint_id, **kwargs):
         sm = get_storage_manager()
         blueprint = sm.get(models.Blueprint, blueprint_id)
 
+        # if finished blueprint validation - cleanup DB entry
+        # and uploaded blueprints folder
+        if blueprint.state == BlueprintUploadState.VALIDATING:
+            uploaded_blueprint_path = '/'.join([
```
Comment (YoniItzhak) [inline rest-service/manager_rest/rest/resources_v3_1/blueprints.py:174]: I meant `os.path.join()`

**Review by YoniItzhak — DISMISSED** [review]

**Review by YoniItzhak — APPROVED** [review]

**Pull request merged**